docs/madr/decisions/006-mesh-traffic-permission.md

```diff
@@ -0,0 +1,785 @@
+# MeshTrafficPermission
+
+* Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/4222
+
+## Context and Problem Statement
+
+[New policy matching MADR](./005-policy-matching.md) introduces a new approach how Kuma applies configuration to proxies.
+Rolling out strategy implies creating a new set of policies that satisfy a new policy matching.
+Current MADR aims to define a MeshTrafficPermission.
+
+## Considered Options
+
+* Create a MeshTrafficPermission policy
+
+## Decision Outcome
+
+Chosen option: create a MeshTrafficPermission policy
+
+### Overview
+
+MeshTrafficPermission allows blocking unwanted incoming traffic for the service.
+There are might be several reasons why it could be useful to block unwanted traffic:
+* following the [principle of least privilege](https://en.wikipedia.org/wiki/Principle_of_least_privilege)
+* cut off access for clients that shouldn't use your service (legacy clients, clients from another zone, etc.)
+
+### Specification
+
+Top-level targetRef can have all available kinds:
+
+```yaml
+targetRef:
+  kind: Mesh|MeshSubset|MeshService|MeshServiceSubset|MeshGatewayRoute|MeshHTTPRoute
+  name: ...
+```
+
+MeshTrafficPermission is an inbound policy that's why it has a "from" array:
+
+```yaml
+from:
+  - targetRef:
+      kind: Mesh|MeshSubset|MeshService|MeshServiceSubset
+      name: ...
+```
+
+Configuration for MeshTrafficPermission has a single field "action" with values "ALLOW", "DENY", 
+"ALLOW_WITH_SHADOW_DENY" and "DENY_WITH_SHADOW_ALLOW".
```

Comment:
what does `DENY_WITH_SHADOW_ALLOW` do?

Reply:
it generates stats as if the request is allowed, but practically it's denied

Reply:
So why would that be useful? The request is denied and I get stats as if it went through, I don't see the value in that. The other way around I get, you allow the request to go through but generate stats as if the action was denied.

Reply:
Isn't it just the dual of that? I mean `ALLOW_WITH_SHADOW_DENY` is useful when you're moving from `ALLOW` -> `DENY`, I think, so `DENY_WITH_SHADOW_ALLOW` is for moving from `DENY` to `ALLOW` . Or put another way, why have shadow allow at all if `DENY_WITH_SHADOW_ALLOW` isn't useful?

Reply:
> DENY_WITH_SHADOW_ALLOW is for moving from DENY to ALLOW

Yeah, probably right. I was under impression that it might be less important (/ not needed at all) since we're going from `DENY` to `ALLOW`. People would be very careful restricting traffic but I thought they might not care that much when loosening up the restrictions but I guess it depends on the change.